draft-ietf-quic-transport.md

```diff
@@ -2849,43 +3014,19 @@ received packets in preference to packets received in the past.
 
 ### ACK Frames and Packet Protection
 
-ACK frames that acknowledge protected packets MUST be carried in a packet that
-has an equivalent or greater level of packet protection.
-
-Packets that are protected with 1-RTT keys MUST be acknowledged in packets that
-are also protected with 1-RTT keys.
-
-A packet that is not protected and claims to acknowledge a packet number that
-was sent with packet protection is not valid.  An unprotected packet that
-carries acknowledgments for protected packets MUST be discarded in its entirety.
+ACK frames MUST only be carried in a packet that has the same packet
```

Comment:
My point is that you can't cross-ACK because you aren't capable of acknowledging something in a different packet number space.

Reply:
I guess this may be a terminology issue. Let me try to explain what I am saying again.
 
Say I receive the following packets 2:1, 2:2, 3:1, 3:3 where X:Y means "epoch X, PN Y". And then I screw up and ACK PN space 3 in epoch 2, so I send epoch=2: ACK[3, 1]. Technically, this means "Acknowledge PNs 3 and 1 in epoch 2", so it's not a cross-ACK from the receiver's perspective, but it *is* an ACK of an unsent packet, which at least at some times we have thought was grounds for tearing down a connection.

Reply:
Yeah, I think that there are two statements to make, each of which differs from the one you chose.

1. An endpoint MUST NOT acknowledge packets you haven't received (and processed).
2. An ACK frame can only acknowledges packets from the same packet number space.

The conjunction of these two things is what you are getting at: if you receive a packet in one packet number space, you cannot acknowledge it in a different space because that would be acknowledging a different packet, one you might not have received.